### PR TITLE
chore(debugging): add timestamp to probe status messages

### PR DESCRIPTION
**Pull request by P403n1x87**
## Description

This change adds the timestamp field to the probe status messages so that they can be ordered in proper chronological order by the backend

## Checklist
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

**Review by shatzi — APPROVED** [review]
Overall I think this would do the trick. However, please see some possible bug in the comment

File: ddtrace/debugging/_probe/status.py [hunk 1]
```diff
@@ -26,10 +26,11 @@ def __init__(self, service, encoder):
         self._encoder = encoder
         self._retry_queue = deque()  # type: deque[Tuple[str, float]]
 
-    def _payload(self, probe, status, message, exc_info=None):
-        # type: (Probe, str, str, Optional[ExcInfoType]) -> str
+    def _payload(self, probe, status, message, timestamp, exc_info=None):
```
[the remark this answers is not in the document]
Comment (P403n1x87) [reply ddtrace/debugging/_probe/status.py:29]: https://docs.python.org/3/library/sys.html#sys.exc_info 🙂

[the remark this answers is not in the document]
Comment (shatzi) [reply ddtrace/debugging/_probe/status.py:29]: 😱 - amazing!

File: ddtrace/debugging/_probe/status.py [hunk 2]
```diff
@@ -76,7 +77,7 @@ def _write(self, probe, status, message, exc_info=None):
                     meter.increment("backlog.buffer_full")
                     return
```
[the remark this answers is not in the document]
Comment (shatzi) [reply ddtrace/debugging/_probe/status.py:78]: Yes. but we never get tot line 80 - so the message sent to _write is never get added to the queue - is that right?

[the remark this answers is not in the document]
Comment (brettlangdon) [reply ddtrace/debugging/_probe/status.py:78]: the item will get picked up for the next payload.

there is the "retry queue", if we try to flush and the buffer ends up getting filled, then the next time we try to write we'll also try to re-add previously failed items from the retry queue.

this flow lgtm at least

File: tests/debugging/probe/test_registry.py [hunk 3]
```diff
@@ -56,6 +56,8 @@ def test_registry_location_error():
     assert not registry.get_pending(__file__)
 
     # Check that we emitted the correct diagnostic error message
+    for e in status_logger.queue:
+        del e["timestamp"]
```
Comment (brettlangdon) [inline tests/debugging/probe/test_registry.py:60]: why? do that we can still assert equality of a list of dicts?

File: ddtrace/debugging/_probe/status.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (P403n1x87) [reply ddtrace/debugging/_probe/status.py:78]: > Yes. but we never get tot line 80 - so the message sent to _write is never get added to the queue - is that right?

Correct. We never get to line 80, during this attempt. Then what @brettlangdon said applies. We flush the buffer periodically, so we hold on to the message and try to put it in the outgoing buffer at the next flush attempt.

File: tests/debugging/probe/test_registry.py (hunk 3 above)
Comment (P403n1x87) [reply tests/debugging/probe/test_registry.py:60]: yep, the timestamp is just randomness here, so we get rid of it before the comparison

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]

**Pull request merged**